File: Floyd’s Cycle Detection (Tortoise and Hare)/Go/Floyd's-Cycle-Detection.go
package main

import "fmt"

type Node struct {
	data int
	next *Node
}

func hasCycle(head *Node) bool {
	slow := head
	fast := head

	for fast != nil && fast.next != nil {
		slow = slow.next
		fast = fast.next.next

		if slow == fast {
			return true
		}
	}
	return false
}

func main() {
	head := &Node{1, nil}
	head.next = &Node{2, nil}
	head.next.next = &Node{3, nil}
	head.next.next.next = &Node{4, nil}
	head.next.next.next.next = head.next // creates cycle

	if hasCycle(head) {
		fmt.Println("Cycle detected")
	} else {
		fmt.Println("No cycle")
	}
}

File: Floyd’s Cycle Detection (Tortoise and Hare)/Python/Floyd's-Cycle-Detection.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

def hasCycle(head):
    slow = fast = head
    while fast and fast.next:
        slow = slow.next
        fast = fast.next.next
        if slow == fast:
            return True
    return False

# Example usage
head = Node(1)
head.next = Node(2)
head.next.next = Node(3)
head.next.next.next = Node(4)
head.next.next.next.next = head.next  # creates cycle

print("Cycle detected" if hasCycle(head) else "No cycle")
